app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,41 @@
+class AuthorsController < ApplicationController
+  def index
+    authors=Author.all
```

Comment:
add filters, name based, email based, premium_user

- should be able to filter all premium_users
- should be able search using email
- should be able to search using name

Reply:
added a filter to fetch only premium user and created a search route to search for user's using name or email, please check latest commit.

Reply:
Changes implemented, please check the latest commit.